src/core/server/http/http_server.ts

```diff
@@ -104,9 +105,16 @@ export class HttpServer {
     return this.server !== undefined && this.server.listener.listening;
   }
 
-  private registerRouter(router: IRouter) {
+  private registerRouter(router: IRouter, allowAfterListening = false) {
     if (this.isListening()) {
-      throw new Error('Routers can be registered only when HTTP server is stopped.');
+      if (!allowAfterListening) {
+        throw new Error('Routers can be registered only when HTTP server is stopped.');
+      }
+
+      this.registeredRouters.add(router);
```

Comment:
Is this necessary? This route will be registered the second time on `start` if the method is called before `start`.